cmd/kapacitord/main.go

```diff
@@ -89,15 +89,31 @@ func (m *Main) Run(args ...string) error {
 
 		signalCh := make(chan os.Signal, 1)
 		signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+		reloadSignal := make(chan os.Signal, 1)
+		signal.Notify(reloadSignal, os.Interrupt, syscall.SIGHUP)
 		m.Logger.Println("I! Listening for signals")
 
 		// Block until one of the signals above is received
-		select {
-		case <-signalCh:
-			m.Logger.Println("I! Signal received, initializing clean shutdown...")
-			go func() {
-				cmd.Close()
-			}()
+	Loop:
+		for {
+			select {
+			case <-signalCh:
+				m.Logger.Println("I! Signal received, initializing clean shutdown...")
+				go func() {
+					cmd.Close()
+				}()
+				break Loop
+			case <-reloadSignal:
+				m.Logger.Println("I! SIGHUP received, Reloading tasks/templates/directory directory...")
+				if err := cmd.Server.LoadService.Load(); err != nil {
+					// TODO: should this be a sufficient condition to shutdown the server?
```

Comment:
> What kinds of errors can LoadService.Load return?

Errors are either from reading files or making requests to the API.

> I would prefer the operation to be atomic, meaning if it fails it logs the failure and then reverts to its previous state.
> That is not really possible since you are going through the API instead of calling into the DB services directly.

I think this might be possible, albeit painful. And I'd have to make some assumptions.
I could figure out what tasks, templates, and handlers would result from a new `Load` operation, save their state temporarily, and then, in the case of a failure, rewrite the old values back through the API.

> I am not sure what we should do here. Thoughts?

I'm very much in the same boat.  The three possibilities I see

1. Best effort, don't crash the server, log the errors
2. Crash the server and treat the files like configuration
3. Don't crash the server, restore to a previous state

I'm inclined to lean towards 2 or 3, but I think that it might be possible to do things to minimize the likelihood of an error.

For example, we could add a `dry-run` query param to the API that attempts to do everything that the regular request to the API would, but without actually committing any changes? I think this should mostly be feasible (although possibly not desirable).

This way we could do a dry run to "validate" the files that are currently there. Or something like this. Essentially a best effort to validate that the entire `Load` operation will be successful, without touching the API.

Alternatively, I think the case could be for 2, in that these files should be treated like configuration and bad configuration should stop the runtime.